apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/SpannerStressTests.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Cloud.Spanner.V1.Logging;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Google.Cloud.Spanner.Data.IntegrationTests
+{
+    [Collection("Spanner Integration Tests")]
+    [PerformanceLog]
+    public class SpannerStressTests : BaseStressTest
+    {
+        private static int s_myCounter = 1;
+        private static readonly string s_guid = Guid.NewGuid().ToString();
+        [ThreadStatic]
+        private static Random s_rnd;
+        private TestDatabaseFixture _testFixture;
+
+        public SpannerStressTests(TestDatabaseFixture testFixture, ITestOutputHelper outputHelper)
+        {
+            _testFixture = testFixture;
+            SpannerOptions.Instance.LogPerformanceTraces = true;
+            TestLogger.TestOutputHelper = outputHelper;
+        }
+
+        protected override async Task<TimeSpan> TestWrite1(Stopwatch sw)
+        {
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                var localCounter = Interlocked.Increment(ref s_myCounter);
+                var insertCommand = connection.CreateInsertCommand(
+                    _testFixture.StressTestTable, new SpannerParameterCollection
+                    {
+                        {"ID", SpannerDbType.String},
+                        {"Title", SpannerDbType.String}
+                    });
+                insertCommand.Parameters["ID"].Value = $"{s_guid}{localCounter}";
+                insertCommand.Parameters["Title"].Value = "Title";
+
+                bool retry = true;
+                long delay = 0;
+                while (retry)
+                {
+                    retry = false;
+                    try
+                    {
+                        //The open is implicit here...
+                        await insertCommand.ExecuteNonQueryAsync();
+                    }
+                    catch (Exception e) when (e.IsTransientSpannerFault())
```

Comment:
We should talk about retries and whether we can push things down a level, perhaps - there's already a RetrySettings at the gRPC/GAPIC level which allows custom predicates. It may not be feasible or desirable, but let's check. (Multiple levels of retry always make me nervous.)